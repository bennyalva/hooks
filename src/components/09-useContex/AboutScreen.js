import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {
    const {user, setUser} = useContext(UserContext);
    const handlerLogout = () => {
        setUser({});
    }
    return (
        <div>
            <h1>AboutScreen</h1>
            <hr />
            <pre>
                { JSON.stringify(user, null, 3)}
            </pre>
            <button onClick={handlerLogout}>
                Logout
            </button>
        </div>
    )
}
