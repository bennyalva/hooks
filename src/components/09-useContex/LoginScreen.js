import React from 'react'
import { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    const {setUser} = useContext(UserContext)
    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            <button
                    onClick={() => setUser({id: 123456, name: 'rey'})}>
                        Login
            </button>
        </div>
    )
}
