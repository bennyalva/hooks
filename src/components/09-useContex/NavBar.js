import React from 'react'
import { Link, NavLink } from 'react-router-dom';
export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-md navbar-light bg-ligth">
            <Link to="/" className="navbar-brand">useContext</Link>
            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink exact activeClassName="active" to="/" className="nav-item nav-link">Home</NavLink>
                    <NavLink exact activeClassName="active" to="./login" className="nav-item nav-link">Login </NavLink>
                    <NavLink exact activeClassName="active" to="./about" className="nav-item nav-link">About </NavLink>
                </div>
            </div>
        </nav>
    )
}
