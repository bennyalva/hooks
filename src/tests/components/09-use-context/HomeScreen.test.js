import React from 'react';
import { UserContext } from '../../../components/09-useContex/UserContext';
const { mount } = require("enzyme");
const { HomeScreen } = require("../../../components/09-useContex/HomeScreen");

describe('Tests on HomeScreen', () => {
    const user = {
        name: 'benny',
        email: 'bennytescha@gmail.com'
    }
    const wrapper = mount(<UserContext.Provider value={{
     user
    }}>
            <HomeScreen />
         </UserContext.Provider>)
   test('should show correctly', () => {
      expect(wrapper).toMatchSnapshot(); 
   });
    
});
