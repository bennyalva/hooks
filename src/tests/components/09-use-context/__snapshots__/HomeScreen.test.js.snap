// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on HomeScreen should show correctly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "benny",
   "email": "bennytescha@gmail.com"
}
  </pre>
</div>
`;
